genie-core/src/test/java/com/netflix/genie/core/jobs/workflow/impl/InitialSetupTaskUnitTest.java

```diff
@@ -0,0 +1,244 @@
+/*
+ *
+ *  Copyright 2015 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.core.jobs.workflow.impl;
+
+import com.netflix.genie.common.dto.Cluster;
+import com.netflix.genie.common.dto.Command;
+import com.netflix.genie.core.jobs.JobConstants;
+import com.netflix.genie.test.categories.UnitTest;
+import com.netflix.spectator.api.Registry;
+import org.apache.commons.io.FileUtils;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.mockito.Mockito;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Writer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.UUID;
+
+/**
+ * Tests for the InitialSetup task.
+ *
+ * @author mprimi
+ * @since 3.1.0
+ */
+@Category(UnitTest.class)
+public class InitialSetupTaskUnitTest {
+
+    private InitialSetupTask initialSetupTask;
+    private Path temporaryTestDirectory;
+
+    /**
+     * Setup to run before each test.
+     * @throws IOException if a temporary folder cannot be created
+     */
+    @Before
+    public void setUp() throws IOException {
+        final Registry registry = Mockito.mock(Registry.class);
+        this.initialSetupTask = new InitialSetupTask(registry);
+        this.temporaryTestDirectory = Files.createTempDirectory(InitialSetupTaskUnitTest.class.getSimpleName() + "-");
```

Comment:
Findbugs (and possibly other static analysis tools) may cry "this method should be private!".
Adding @VisibleForTesting tells them the reason it's not private:
```
/**
 * Annotates a program element that exists, or is more widely visible than
 * otherwise necessary, only for use in test code.
 */
public @interface VisibleForTesting {
}
```